addons/hw_drivers/tools/helpers.py

```diff
@@ -133,33 +153,45 @@ def check_git_branch():
     http = urllib3.PoolManager(cert_reqs='CERT_NONE')
     try:
-        response = http.request('POST',
+        response = http.request(
+            'POST',
             server + "/web/webclient/version_info",
             body='{}',
-            headers={'Content-type': 'application/json'}
+            headers={'Content-type': 'application/json'},
         )
 
         if response.status == 200:
             git = ['git', '--work-tree=/home/pi/odoo/', '--git-dir=/home/pi/odoo/.git']
 
             db_branch = json.loads(response.data)['result']['server_serie'].replace('~', '-')
-            if not subprocess.check_output(git + ['ls-remote', 'origin', db_branch]):
-                db_branch = 'master'
-
-            local_branch = subprocess.check_output(git + ['symbolic-ref', '-q', '--short', 'HEAD']).decode('utf-8').rstrip()
-            _logger.info("Current IoT Box local git branch: %s / Associated Odoo database's git branch: %s", local_branch, db_branch)
+            local_branch = (
+                subprocess.check_output(git + ['symbolic-ref', '-q', '--short', 'HEAD']).decode('utf-8').rstrip()
+            )
+            _logger.info(
+                "Current IoT Box local git branch: %s / Associated Odoo database's git branch: %s",
+                local_branch,
+                db_branch,
+            )
 
             if db_branch != local_branch:
                 with writable():
-                    subprocess.check_call(["rm", "-rf", "/home/pi/odoo/addons/hw_drivers/iot_handlers/drivers/*"])
-                    subprocess.check_call(["rm", "-rf", "/home/pi/odoo/addons/hw_drivers/iot_handlers/interfaces/*"])
-                    subprocess.check_call(git + ['branch', '-m', db_branch])
-                    subprocess.check_call(git + ['remote', 'set-branches', 'origin', db_branch])
-                    os.system('/home/pi/odoo/addons/point_of_sale/tools/posbox/configuration/posbox_update.sh')
-
+                    _logger.info("Creating a backup of the current odoo folder")
+                    subprocess.run(
+                        ["rsync", "-av", "--delete", "/home/pi/odoo/", "/home/pi/.odoo.backup/"],
+                        check=False,
+                        stdout=subprocess.DEVNULL,
+                    )  # create a backup of the current odoo folder
+                    subprocess.run(git + ['branch', '-m', db_branch], check=True)
+                    subprocess.run(git + ['remote', 'set-branches', 'origin', db_branch], check=True)
+                    _logger.info("Updating odoo folder to the branch %s", db_branch)
+                    subprocess.run(
+                        ['/home/pi/odoo/addons/point_of_sale/tools/posbox/configuration/posbox_update.sh'],
+                        check=False,
+                    )
     except Exception as e:
         _logger.error('Could not reach configured server')
-        _logger.error('A error encountered : %s ', e)
+        _logger.error('An error occurred : %s ', e)
```

Comment:
this errors messages are terrible, could you replace them with a good `_logger.exception` instead please ?